### DISCUSSION TITLE
What is the Use Case for Using `*args` as a Type Variable Tuple?

### DISCUSSION THREAD

**Discussion by adam-grant-hendry** [microsoft/pyright#3957] [Q&A]
I don't understand the intended use case for [`*args` as a Type Variable Tuple](https://peps.python.org/pep-0646/#args-as-a-type-variable-tuple) as laid out in `PEP 646`.

I don't understand why I wouldn't do [what's recommended by typeshed](https://typing.readthedocs.io/en/latest/source/best_practices.html#using-any-and-object) and give `*args` the type `object` so it can be any type and then type narrow in the function body. It seems like the target method that processes `*args` is going to have to type narrow any way if its items are truly indeterminate in size and type.

Maybe it is used for top-level functions that use helper functions that accept `*args`? The `PEP` provides `os.execle` as an example, which [passes `args` to `os.execve`](https://github.com/python/cpython/blob/78359b1d45608439f8e03b8e86174fe7b04d3e08/Lib/os.py#L543), but `typeshed` seems to [manage with using `Any`](https://github.com/python/typeshed/blob/20bc8fa4e30df5f879606dfc37dd0b4bcebaeedd/stdlib/os/__init__.pyi#L795) (granted, `TypeVarTuple` isn't supported by `mypy` yet). Again though, I would have just typed this as `object`.

What is the intended use case for [`*args` as a Type Variable Tuple](https://peps.python.org/pep-0646/#args-as-a-type-variable-tuple)?

**Comment by erictraut** [comment]
This question would be more appropriate for the python/typing discussion forum. The authors of PEP 646 follow the posts in that forum, but they're unlikely to see this post.

**Reply by adam-grant-hendry** [reply]
Thank you, I've repeated my question [there](https://github.com/python/typing/discussions/1258).

**Reply by adam-grant-hendry** [reply]
On a side note, in your professional opinion, does it make sense to you where you would use [`*args` as a Type Variable Tuple](https://peps.python.org/pep-0646/#args-as-a-type-variable-tuple)?
